frontend/src/context/FolderContext.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { createContext, useContext, useState, ReactNode } from 'react';
+import { Folder } from '../utils/types';
+
+interface FolderContextType {
+  selectedFolder: Folder | null;
+  selectFolder: (folder: Folder | null) => void;
+}
+
+const FolderContext = createContext<FolderContextType | undefined>(undefined);
+
```

Comment:
is undefined in <> necessary?

Reply:
I think it is, initially after accessing the site there is no folder context provided. But I think it won't be a big issue, because I'm planning to redesign those file and folder contexts when implementing full crud on file and folders